code/modules/power/cable.dm

```diff
@@ -415,3 +412,202 @@
 		logTheThing(LOG_STATION, user, "lays a cable[powered == 1 ? " (powered when connected)" : ""] at [log_loc(src)].")
 
 	return
+
+/// a cable spawner which can spawn multiple cables to connect to other cables around it.
+/obj/cablespawner
+	name = "power cable spawner"
+	icon = 'icons/obj/power_cond.dmi'
+	icon_state = "superstate"
+	layer = CABLE_LAYER
+	plane = PLANE_NOSHADOW_BELOW
+	color = "#DD0000"
+	// this would make it connect to the centre, for like terminals and whatnot
+	// has to be var edited because lazy
+	var/override_centre_connection = FALSE
+	var/cable_type = /obj/cable
+	/// the following 8 bit variable constants use bit flags to tell which 8 tiles need connections
+	/*
+	* bitflag of the tiles surrounding itself:
+	* i.e.(10)(9)(6)(5) 8  4  2  1 (normal bitflags) (used when you only need one direction)
+	* i.e. SW NW SE NE  W  E  S  N (corresponding directions)
+	*     128 64 32 16  8  4  2  1 (actual bitflags) (used for multiple directions)
+	* i think someone called it 8 bit directions once? idk
+	* anyway thats what cable_surr does
+	*/
+	var/const/SOUTHWEST_UNIQUE = 128
+	var/const/NORTHWEST_UNIQUE = 64
+	var/const/SOUTHEAST_UNIQUE = 32
+	var/const/NORTHEAST_UNIQUE = 16
+	// these constants ARE needed they are DIFFERENT from the regular NORTHWEST and such
+	var/cable_surr = 0
+/// reinforced, thick cables. They should also connect to the regular kind.
+/obj/cablespawner/reinforced
+	name = "reinforced power cable spawner"
+	icon = 'icons/obj/power_cond.dmi'
+	icon_state = "superstate"
+	cable_type = /obj/cable/reinforced
+	color = "#075C90"
+
+/// creates new cablespawners
+/obj/cablespawner/New()
+	..()
```

Comment:
```suggestion
```
no need to define `New` if it just calls parent

Reply:
ah, good catch. That was from before i heard about `initialize()`